src/Admin/AbstractAdmin.php

```diff
@@ -1570,10 +1586,22 @@ public function setFilterPersister(?FilterPersisterInterface $filterPersister =
     }
 
     /**
+     * NEXT_MAJOR: Remove this function.
+     *
+     * @deprecated since since sonata-project/sonata-admin-bundle 3.x, to be remove in 4.0.
+     *
      * @param int $maxPerPage
      */
     public function setMaxPerPage($maxPerPage)
     {
+        @trigger_error(
+            sprintf(
+                'The method %s is deprecated since sonata-project/sonata-admin-bundle 3.x and will be removed in 4.0.',
+                __METHOD__
+            ),
+            E_USER_DEPRECATED
+        );
+
```

Comment:
```suggestion
        @trigger_error(sprintf(
            'The method %s is deprecated since sonata-project/sonata-admin-bundle 3.x and will be removed in 4.0.',
            __METHOD__
        ), E_USER_DEPRECATED);

```